2023-04-29 00:32:37,177 ERROR: Exception on / [GET] [in C:\VSCode\microblog\venv\Lib\site-packages\flask\app.py:1454]
Traceback (most recent call last):
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask\app.py", line 2070, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask\app.py", line 1515, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask\app.py", line 1513, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask\app.py", line 1499, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\app\routes.py", line 41, in index
    return render_template('index.html', title='Home Page', form=form, posts=posts.items, next_url=next_url, prev_url=prev_url)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask\templating.py", line 147, in render_template
    return _render(
           ^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask\templating.py", line 128, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\jinja2\environment.py", line 1304, in render
    self.environment.handle_exception()
  File "C:\VSCode\microblog\venv\Lib\site-packages\jinja2\environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\VSCode\microblog\app\templates\index.html", line 2, in <module>
    {% import 'bootstrap/wtf.html' as wtf %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\app\templates\base.html", line 1, in <module>
    {% extends 'bootstrap/base.html' %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 1, in <module>
    {% block doc -%}
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 4, in <module>
    {%- block html %}
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 20, in <module>
    {% block body -%}
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 21, in <module>
    {% block navbar %}
  File "C:\VSCode\microblog\app\templates\base.html", line 17, in <module>
    <a class="navbar-brand" href="{{ url_for('main.index') }}">Microblog</a>
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask\helpers.py", line 338, in url_for
    return appctx.app.handle_url_build_error(error, endpoint, values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask\helpers.py", line 325, in url_for
    rv = url_adapter.build(
         ^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\werkzeug\routing.py", line 2305, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.BuildError: Could not build url for endpoint 'main.index'. Did you mean 'index' instead?
2023-04-29 00:39:03,801 INFO: Microblog startup [in C:\VSCode\microblog\app\__init__.py:51]
2023-04-29 00:39:06,475 ERROR: Exception on / [GET] [in C:\VSCode\microblog\venv\Lib\site-packages\flask\app.py:1454]
Traceback (most recent call last):
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask\app.py", line 2070, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask\app.py", line 1515, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask\app.py", line 1513, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask\app.py", line 1499, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\app\routes.py", line 41, in index
    return render_template('index.html', title='Home Page', form=form, posts=posts.items, next_url=next_url, prev_url=prev_url)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask\templating.py", line 147, in render_template
    return _render(
           ^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask\templating.py", line 128, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\jinja2\environment.py", line 1304, in render
    self.environment.handle_exception()
  File "C:\VSCode\microblog\venv\Lib\site-packages\jinja2\environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\VSCode\microblog\app\templates\index.html", line 2, in <module>
    {% import 'bootstrap/wtf.html' as wtf %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\app\templates\base.html", line 1, in <module>
    {% extends 'bootstrap/base.html' %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 1, in <module>
    {% block doc -%}
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 4, in <module>
    {%- block html %}
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 20, in <module>
    {% block body -%}
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 21, in <module>
    {% block navbar %}
  File "C:\VSCode\microblog\app\templates\base.html", line 17, in <module>
    <a class="navbar-brand" href="{{ url_for('main.index') }}">Microblog</a>
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask\helpers.py", line 338, in url_for
    return appctx.app.handle_url_build_error(error, endpoint, values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\flask\helpers.py", line 325, in url_for
    rv = url_adapter.build(
         ^^^^^^^^^^^^^^^^^^
  File "C:\VSCode\microblog\venv\Lib\site-packages\werkzeug\routing.py", line 2305, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.BuildError: Could not build url for endpoint 'main.index'. Did you mean 'index' instead?
2023-04-29 00:43:45,471 INFO: Microblog startup [in C:\VSCode\microblog\app\__init__.py:51]
